### PR TITLE
avoid resizing filesystem for block volumes

### LINKED ISSUE #111: node staging fails for block volumes
Volume resizing fails for block volumes:

`
 Jan 23 16:38:19.572: INFO: At 2023-01-23 16:28:34 +0000 UTC - event for inline-volume-tester-tw978: {kubelet ci-op-p95ft5gl-3565d-mbqkm-worker-2-xhdlf} VolumeResizeFailed: NodeExpandVolume.NodeExpandVolume failed for volume "pvc-43193e9e-e642-48cd-9088-7ee4c57eb081" : Expander.NodeExpand failed to expand the volume : rpc error: code = Internal desc = {RequestID: 8ca47df4-4c62-4bdc-a58f-5b127e628a8b , Code: FileSystemResizeFailed, Description: Failed to resize the file system, BackendError: failed to read size of device devtmpfs: exit status 1: blockdev: cannot open devtmpfs: No such file or directory, Action: Please check if there is any error in PVC describe related with volume resize} 
`

The root cause is that `NodeStageVolume` gRPC does not check if a volume is block device. So if the volume *is* a block device it evaluates `volumePath` incorrectly. In the event shown above it's set to `devtmpfs` which is not a real path to device. When this value is passed to resizer (`NeedResize` function specifically) which tries to find the device so it can compare block and fs sizes - it fails with `No such file or directory`.

This is currently causing our Openshift CI to fail when rebasing this driver downstream: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_ibm-vpc-block-csi-driver/30/pull-ci-openshift-ibm-vpc-block-csi-driver-master-e2e-ibmcloud-csi/1617543897712431104

### PR DESCRIPTION
**Pull request by RomanBednar**
Resolves #111

### CODE REVIEW

File: pkg/ibmcsidriver/node.go [hunk 1]
```diff
@@ -473,6 +478,36 @@ func (csiNS *CSINodeServer) NodeExpandVolume(ctx context.Context, req *csi.NodeE
...
+			return nil, status.Errorf(codes.Internal, "failed to determine if volumePath [%v] is a block device: %v", volumePath, err)
+		}
+		if isBlock {
+			// Skip resizing for Block NodeExpandVolume
+			capacity, err := csiNS.Stats.DeviceInfo(volumePath)
+			if err != nil {
+				return nil, status.Errorf(codes.Internal, "failed to get block capacity on path %s: %v", volumePath, err)
+			}
+			klog.V(4).InfoS("NodeExpandVolume: called, since given volumePath is a block device, ignoring...", "volumeID", volumeID, "volumePath", volumePath)
+			return &csi.NodeExpandVolumeResponse{CapacityBytes: capacity}, nil
```
Comment (jsafrane) [inline pkg/ibmcsidriver/node.go:507]: Why this NodeExpandVolumeResponse has CapacityBytes and the other for `// Noop for block NodeExpandVolume` does not? Both are block volumes and IMO should have the same response.

Comment (RomanBednar) [reply pkg/ibmcsidriver/node.go]: This could be written better, see the new version.

**Comment by ambiknai** [comment]
@RomanBednar Thanks for the PR. I was able to recreate the issue with the explanation you provided. I was trying to deploy this code and see how it behaves but unfortunately looks like code base is not up to date. Could you please rebase with master once. I will build custom image and test on latest code.

**Comment by ambiknai** [comment]
> @RomanBednar Thanks for the PR. I was able to recreate the issue with the explanation you provided. I was trying to deploy this code and see how it behaves but unfortunately looks like code base is not up to date. Could you please rebase with master once. I will build custom image and test on latest code.

Please ignore above ^^^ My local code base was messed up. I am able to get the containers up and running now.

**Review by ambiknai — COMMENTED** [review]
Could you please make  this change in DeviceInfo method =>  `--getsize64`
While testing I got this error
```
  Warning  VolumeResizeFailed      3m59s (x32 over 52m)  kubelet                  NodeExpandVolume.NodeExpandVolume failed for volume "pvc-a6e30c46-958f-4f70-a740-4ebb3bb6837f" : Expander.NodeExpand failed to expand the volume : rpc error: code = Internal desc = failed to get block capacity on path /var/data/kubelet/plugins/kubernetes.io/csi/volumeDevices/publish/pvc-a6e30c46-958f-4f70-a740-4ebb3bb6837f/faffab22-d11c-4c59-86f7-e07693d655e8: failed to get size of block volume at path /var/data/kubelet/plugins/kubernetes.io/csi/volumeDevices/publish/pvc-a6e30c46-958f-4f70-a740-4ebb3bb6837f/faffab22-d11c-4c59-86f7-e07693d655e8: output: blockdev: cannot open getsize64: No such file or directory
  ```

**Comment by RomanBednar** [comment]
> Could you please make this change in DeviceInfo method => `--getsize64` While testing I got this error

@ambiknai Good catch, posting the fix now.

**Review by arahamad — APPROVED** [review]
/lgtm

please verify changes as well

**Pull request merged**